File: python/nivel 5 simulacion/27.py
"""
Ejercicio 27
Validar una contraseña (largo mínimo, números, letras)
"""
password = input("Ingresa una contraseña: ")

tiene_letra = False
tiene_numero = False

# primera confirmacion
if len(password) < 8:
  print("Contraseña invalida: debe de tener al menos 8 digitos.")
else:
  # segunda y tercera confirmacion
  for c in password:
    if c.isalpha():
      tiene_letra = True
    elif c.isdigit():
      tiene_numero = True

  if tiene_letra and tiene_numero:
    print("Contraseña Valida ")
  else:
    print("Contraseña Invalida: debe contener al menos una letra y un numero")

File: javascript/nivel 5 simulacion/27.js
/*
Ejercicio 27
Validar una contraseña (largo mínimo, números y letras)
*/
let password = prompt("Ingresa una contraseña: ")

let tieneLetra = false;
let tieneNumero = false;

//Regla 1: largo minimo
if(password.length < 8){
  console.log("Contraseña invalidad, debe tener al menos 8 caracteres, entre numeros y letras");
}else{
 //regla 2 y 3
  for(let c of password){
      if(c >= 'a' && c <='z' || c >= 'A' && c <= 'Z'){
        tieneLetra = true;
      }else if( c >='0' && c <= '9'){
        tieneNumero = true;
      }
  }

  if(tieneLetra && tieneNumero){
    console.log("Contraseña Valida");
  }else{
    console.log("Contraseña invalida");
  }
}